src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,626 @@
+
+  etotal = haero::min(etotal, one);
+} // calc_impact_efficiency
+
+/*=====================================================================*/
+KOKKOS_INLINE_FUNCTION
+void calc_1_impact_rate(const Real dg0,     //  in
+                        const Real sigmag,  //  in
+                        const Real rhoaero, //  in
+                        const Real temp,    //  in
+                        const Real press,   //  in
+                        Real &scavratenum,  // out
+                        Real &scavratevol)  // out
+
+{
+  // this subroutine computes a single impaction scavenging rate
+  //  for precipitation rate of 1 mm/h
+
+  //   function parameters
+  // @param [in]  dg0         geometric mean diameter of aerosol
+  // @param [in]  sigmag      geometric standard deviation of
+  // size distribution [cm]
+  // @param [in]  rhoaero     aerosol density [g/cm^3]
+  // @param [in]  temp        temperature [K] real(r8),
+  // @param [in]  press       pressure [dyne/cm^2]
+  // @param [out] scavratenum scavenging rate for aerosol number [1/hour]
+  // @param [out] scavratenum scavenging rate for aerosol volume [1/hour]
+
+  // const Real SHR_CONST_BOLTZ =
+  //     1.38065e-23; //  ! Boltzmann's constant ~ J/K/molecule
+  // const Real SHR_CONST_AVOGAD =
+  //     6.02214e26; //   ! Avogadro's number ~ molecules/kmole
+  // const Real rgas_kmol =
+  //     SHR_CONST_AVOGAD *
+  //     SHR_CONST_BOLTZ; //       ! Universal gas constant ~ J/K/kmole => ! Gas
+  // //                      //       constant (J/K/mol)
+  // const Real rgas = rgas_kmol * 1.e-3 * 1.e7; //
+  // // air molar density [dyne/cm^2/erg*mol = mol/cm^3]
+  // const Real cair = press / (rgas * temp);
+  // air molar density [mol/cm^3]
+  // const Real rhoair = 28.966 * cair;
+
+  const Real pi = haero::Constants::pi;
+  const Real zero = 0;
+  const Real one = 1;
+  const Real two = 2;
+  const Real ten = 10;
+  const Real one_thousand = 1000;
+
+  // local variables
+  Real rrainsv[nrainsvmax] = {zero};    // rain radius for each bin [cm]
+  Real xnumrainsv[nrainsvmax] = {zero}; // rain number for each bin [#/cm3]
+  Real vfallrainsv[nrainsvmax] = {
+      zero}; // rain falling velocity for each bin [cm/s]
+
+  Real raerosv[naerosvmax] = {zero}; // aerosol particle radius in each bin [cm]
+  Real fnumaerosv[naerosvmax] = {
+      zero}; // fraction of total number in the bin [fraction]
+  Real fvolaerosv[naerosvmax] = {
+      zero}; // fraction of total volume in the bin [fraction]
+
+  // this subroutine is calculated for a fix rainrate of 1 mm/hr
+  // precipitation rate, fix as 1 mm/hr in this subroutine [cm/s]
+  const Real precip = one / Real(36000.); //  1 mm/hr in cm/s
+
+  // set the iteration radius for rain droplet
+  // rain droplet bin information [cm]
+  const Real rlo = .005;
+  const Real rhi = .250;
+  const Real dr = 0.005;
+  // Nearest whole number: nint
+  // number of rain bins
+  const int nr = Real(1) + int((rhi - rlo) / dr);
```

Comment:
would `int(1) + round((rhi - rlo) / dr)` make more sense here, since the fortran code is using `nint()` (nearest integer)?


Alternatively: `const in nr = 50;` 🥸

Reply:
Good point.